java/system-test/system-test-data-generation/src/main/java/sleeper/systemtest/datageneration/WriteRandomDataDirect.java

```diff
@@ -37,27 +47,43 @@ private WriteRandomDataDirect() {
 
     public static void writeWithIngestFactory(
             SystemTestPropertyValues systemTestProperties, InstanceIngestSession session) throws IOException {
+        StateStoreProvider stateStoreProvider = session.createStateStoreProvider();
         writeWithIngestFactory(
                 IngestFactory.builder()
                         .objectFactory(ObjectFactory.noUserJars())
                         .localDir("/mnt/scratch")
-                        .stateStoreProvider(session.createStateStoreProvider())
+                        .stateStoreProvider(stateStoreProvider)
                         .instanceProperties(session.instanceProperties())
                         .hadoopConfiguration(session.hadoopConfiguration())
                         .s3AsyncClient(session.s3Async())
                         .build(),
+                addFilesToStateStore(session.sqs(), session.instanceProperties(), session.tableProperties(), stateStoreProvider),
                 systemTestProperties, session.tableProperties());
     }
 
     public static void writeWithIngestFactory(
-            IngestFactory ingestFactory, SystemTestPropertyValues properties, TableProperties tableProperties) throws IOException {
+            IngestFactory ingestFactory, AddFilesToStateStore addFilesToStateStore,
+            SystemTestPropertyValues testProperties, TableProperties tableProperties) throws IOException {
         CloseableIterator<Record> recordIterator = new WrappedIterator<>(
-                WriteRandomData.createRecordIterator(properties, tableProperties));
+                WriteRandomData.createRecordIterator(testProperties, tableProperties));
```

Comment:
Don't we still need to close this record iterator?

Reply:
No, there's nothing to close.

Reply:
Actually, it looks like the WrappedIterator isn't needed at all. I'll remove it, so it's not closeable in the first place.